examples/04-Retrieval-Model.ipynb

```diff
@@ -0,0 +1,1866 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "c6607333",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/usr/local/lib/python3.8/dist-packages/cudf/core/dataframe.py:1253: UserWarning: The deep parameter is ignored and is only included for pandas compatibility.\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "user_id = [\"user_id\"] >> Categorify() >> TagAsUserID()\n",
+    "item_id = [\"item_id\"] >> Categorify() >> TagAsItemID()\n",
+    "\n",
+    "item_features = [\"item_category\", \"item_shop\", \"item_brand\"] >> nvt.ops.Categorify() >> AddTags(tags=[Tags.ITEM])\n",
+    "\n",
+    "user_features = ['user_shops', 'user_profile', 'user_group', \n",
+    "       'user_gender', 'user_age', 'user_consumption_2', 'user_is_occupied',\n",
+    "       'user_geography', 'user_intentions', 'user_brands', 'user_categories'] \\\n",
+    "        >> nvt.ops.Categorify() >> AddTags(tags=[Tags.USER])\n",
+    "\n",
+    "inputs = user_id + item_id + item_features + user_features + ['click'] \n",
+    "\n",
+    "outputs = inputs >> Filter(f=lambda df: df[\"click\"] == 1)\n",
+    "\n",
+    "workflow_fit_transform(outputs, train_path, test_path, output_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "506dd46e",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Building Two-Tower Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2556f5b8",
+   "metadata": {},
+   "source": [
+    "We will use Two-Tower Model for Item retrieval task. Real-world large scale recommender systems have hundreds of millions of items (products) and users. It is Thus, these systems are often composed of two stages: candidate generation (retrieval) and ranking (scoring the retrieved items). You can read more about two stage Recommender Systems [here](https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/45530.pdf). In this example, we're going to focus on the retrieval stage.\n",
+    "\n",
+    "A Two-Tower Model consists of item (candidate) and user (query) encoder towers. With two towers, the model can learn representations (embeddings) for queries and candidates separately. \n",
+    "\n",
+    "<img src=\"./images/TwoTower.png\"  width=\"30%\">\n",
+    "\n",
+    "Image Adapted from: [Off-policy Learning in Two-stage Recommender Systems](https://dl.acm.org/doi/abs/10.1145/3366423.3380130)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "76908dad",
+   "metadata": {},
+   "source": [
+    "We use the `schema` object to define our model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "363ca210",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "schema = TensorflowMetadata.from_proto_text_file('/workspace/retrieval/processed/train/').to_merlin_schema()"
```

Comment:
I'm sorta wondering if we should just move the code to load the train/valid Datasets up here, and then just get the schema by `train.schema` ? 

```suggestion
    "schema = TensorflowMetadata.from_proto_text_file('/workspace/retrieval/processed/train/').to_merlin_schema()"
```

Reply:
let me try it out.